src/opts.rs

```diff
@@ -89,6 +89,25 @@ pub fn enable_caching(enabled: bool) {
     debug_assert!(error >= 0);
 }
 
+/// Set the maximum data size for the given type of object to be considered
+/// eligible for caching in memory.  Setting to value to zero means that that
+/// type of object will not be cached.  Defaults to 0 for [`ObjectType::Blob`]
+/// (i.e. won't cache blobs) and 4k for [`ObjectType::Commit`],
+/// [`ObjectType::Tree`], and [`ObjectType::Tag`].
+pub fn set_cache_object_limit(kind: ObjectType, size: libc::size_t) {
+    crate::init();
+    let error = unsafe {
+        raw::git_libgit2_opts(
+            raw::GIT_OPT_SET_CACHE_OBJECT_LIMIT as libc::c_int,
+            kind as libc::c_int,
+            size,
+        )
+    };
+    // This function cannot actually fail, but the function has an error return
+    // for other options that can.
+    debug_assert!(error >= 0);
```

Comment:
I don't think this is entirely correct, as I believe this can fail if it is given an invalid type. That shouldn't happen since our `ObjectType` should stay in sync with libgit2. Can this comment be updated to clarify that?